account_invoice_change_currency/tests/test_account_invoice_change_currency.py

```diff
@@ -0,0 +1,334 @@
+# Copyright 2018 Komit <http://komit-consulting.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+import odoo.tests.common as common
+from odoo import fields
+from odoo.tools import float_compare
+
+
+class TestAccountInvoiceChangeCurrency(common.TransactionCase):
+    def setUp(self):
+        super(TestAccountInvoiceChangeCurrency, self).setUp()
+
+        self.precision = self.env["decimal.precision"].precision_get("Payment Terms")
+        res_users_account_manager = self.env.ref("account.group_account_manager")
+        self.manager = (
+            self.env["res.users"]
+            .with_context({"no_reset_password": True})
+            .create(
+                dict(
+                    name="Adviser",
+                    company_id=self.env.ref("base.main_company").id,
+                    login="fm",
+                    email="accountmanager@yourcompany.com",
+                    groups_id=[(6, 0, [res_users_account_manager.id])],
+                )
+            )
+        )
+
+        # Needed to create invoice
+        self.account_type1 = self.env["account.account.type"].create(
+            {
+                "name": "acc type test 1",
+                "type": "receivable",
+                "include_initial_balance": True,
+                "internal_group": "asset",
+            }
+        )
+        self.account_type2 = self.env["account.account.type"].create(
+            {
+                "name": "acc type test 2",
+                "type": "other",
+                "include_initial_balance": True,
+                "internal_group": "asset",
+            }
+        )
+        self.account_account = self.env["account.account"].create(
+            {
+                "name": "acc test",
+                "code": "X2020",
+                "user_type_id": self.account_type1.id,
+                "reconcile": True,
+            }
+        )
+        self.account_account_line = self.env["account.account"].create(
+            {
+                "name": "acc inv line test",
+                "code": "X2021",
+                "user_type_id": self.account_type2.id,
+                "reconcile": True,
+            }
+        )
+        self.product_1 = self.env["product.product"].create({"name": "Product 1"})
+        self.product_2 = self.env["product.product"].create({"name": "Product 2"})
+        self.analytic_account = self.env["account.analytic.account"].create(
+            {"name": "test account"}
+        )
+        self.tax_account = self.env["account.account"].search(
+            [
+                (
+                    "user_type_id",
+                    "=",
+                    self.env.ref("account.data_account_type_current_assets").id,
+                )
+            ],
+            limit=1,
+        )
+
+    def create_journal_type(self, type_string=None):
+        if not type_string:
```

Comment:
Since the default value is a string, it may be specified as method default param